scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -700,6 +700,30 @@ class ScioContext private[scio] (
   ): SCollection[U] =
     applyTransform(Option(name), root)
 
+  def transform[U](f: ScioContext => SCollection[U]): SCollection[U] = transform(this.tfName)(f)
+
+  def transform[U](name: String)(f: ScioContext => SCollection[U]): SCollection[U] =
+    wrap(transform_(name)(f(_).internal))
+
+  private[scio] def transform_[U <: POutput](f: ScioContext => U): U =
+    transform_(tfName)(f)
+
+  private[scio] def transform_[U <: POutput](name: String)(f: ScioContext => U): U = {
+    val sc: ScioContext = this
```

Comment:
are you sure the context can be accessed as a closure ? IIRC it can throw `NotSerializableException`